Userland/Libraries/LibJS/Runtime/SafeStringBuilder.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/SafeStringBuilder.h>
+
+namespace JS {
+
+SafeStringBuilder::SafeStringBuilder(VM& vm)
+    : m_vm(vm)
+{
+}
+
+ThrowCompletionOr<void> SafeStringBuilder::append(char ch)
+{
+    if (try_append(ch).is_error())
+        return m_vm.throw_completion<RangeError>(ErrorType::NotEnoughMemoryToAllocate, length() + 1);
+    return {};
+}
+
+ThrowCompletionOr<void> SafeStringBuilder::append(StringView string)
+{
+    if (try_append(string).is_error())
+        return m_vm.throw_completion<RangeError>(ErrorType::NotEnoughMemoryToAllocate, length() + string.length());
+    return {};
+}
+
+ThrowCompletionOr<void> SafeStringBuilder::append(Utf16View string)
+{
+    if (try_append(string).is_error())
+        return m_vm.throw_completion<RangeError>(ErrorType::NotEnoughMemoryToAllocate, length() + string.length_in_code_units());
```

Comment:
This doesn't seem right, the code units span holds `u16`s, which are two bytes each?

https://github.com/SerenityOS/serenity/blob/3b682e885e492e24aef952050998a2cf1a83f16e/AK/Utf16View.h#L114